src/Dofs/ConstraintHandler.jl

```diff
@@ -1762,3 +1821,146 @@ function _condense_local!(
+        qr_order = _default_bc_qr_order(dbc.qr_order, interpolation)
+        fqr = FacetQuadratureRule{getrefshape(interpolation)}(qr_order)
+        fv = FacetValues(fqr, interpolation, geometric_interpolation(CT))
+        local_facet_dofs, local_facet_dofs_offset =
+            _local_facet_dofs_for_bc(get_base_interpolation(interpolation), 1, 1, field_offset(sdh, field_idx), dirichlet_facetdof_indices)
+        facet_dofs = ArrayOfVectorViews(local_facet_dofs_offset, local_facet_dofs, LinearIndices(1:(length(local_facet_dofs_offset) - 1)))
+
+        filtered_dbc = WeakDirichlet(dbc.f, filtered_set, dbc.field_name, qr_order, fv, facet_dofs)
+
+        _add!(ch, filtered_dbc, facet_dofs)
+
+        dbc_added = true
+    end
+    dbc_added || error("No overlap between dbc::Dirichlet and fields in the ConstraintHandler's DofHandler")
+    return ch
+end
+
+function _add!(ch::ConstraintHandler, dbc::WeakDirichlet, facet_dofs)
+    # loop over all the faces in the set and add the global dofs to `constrained_dofs`
+    constrained_dofs = Int[]
+    cc = CellCache(ch.dh, UpdateFlags(; nodes = false, coords = false, dofs = true))
+    for (cellidx, facetidx) in dbc.facets
+        reinit!(cc, cellidx)
+        local_dofs = facet_dofs[facetidx]
+        for d in local_dofs
+            push!(constrained_dofs, cc.dofs[d])
+        end
+    end
+
+    # save it to the ConstraintHandler
+    push!(ch.idbcs, dbc)
+    for d in constrained_dofs
+        add_prescribed_dof!(ch, d, NaN, nothing)
+    end
+    return ch
+end
+
+
+function _update_weak_dbc!(
+        inhomogeneities::Vector{T}, f::Function, facets::AbstractVecOrSet{FacetIndex}, fv::FacetValues, facet_dofs::ArrayOfVectorViews,
+        dh::AbstractDofHandler, dofmapping::Dict{Int, Int}, dofcoefficients::Vector{Union{Nothing, DofCoefficients{T}}}, time::Real
+    ) where {T}
+    ip = function_interpolation(fv)
+    n = maximum(length, dirichlet_facetdof_indices(ip))
+    Kᶠ = zeros(n, n)
+    aᶠ = zeros(n)
+    fᶠ = zeros(n)
+    for fc in FacetIterator(dh, facets)
+        reinit!(fv, fc)
+        shape_nrs = dirichlet_facetdof_indices(ip)[getcurrentfacet(fv)]
+        solve_weak_dbc!(aᶠ, Kᶠ, fᶠ, f, fv, shape_nrs, getcoordinates(fc), time)
+        for (idof, shape_nr) in enumerate(shape_nrs)
+            globaldof = celldofs(fc)[shape_nr]
+            dbc_index = dofmapping[globaldof]
+            # Only DBC dofs are currently update!-able so don't modify inhomogeneities
+            # for affine constraints
+            if dofcoefficients[dbc_index] === nothing
+                inhomogeneities[dbc_index] = aᶠ[idof]
+            end
+        end
+    end
+    return nothing
+end
+
+function solve_weak_dbc!(aᶠ, Kᶠ, fᶠ, bc_fun, fv, shape_nrs, cell_coords, time)
+    fill!(Kᶠ, 0)
+    fill!(fᶠ, 0)
+    # Support varying number of facetdofs (for ref shapes with different facet types)
+    for i in (length(shape_nrs) + 1):size(Kᶠ, 1)
+        Kᶠ[i, i] = 1
+    end
```

Comment:
Currently, we don't have any H(div) or H(curl) interpolations with different number of shape functions associated with different facets. But theoretically, we could, hence this addition which is not tested. 

Removing it will case a singular matrix if such are implemented in the future, alternatively we could check and throw if `size(K,1) != length(shape_nrs)`.

Reply:
We also have the very same issue with H1/L2 interpolations and this property, so let us not introduce this assumption for now in parts of the code.